tclx/data/datasets.py

```diff
@@ -0,0 +1,150 @@
+from torch.utils.data import Dataset
+import torch
+from tqdm import tqdm
+import numpy as np
+
+
+# Anthropic fine-tunes language model on entire dialogue, not just responses
+class SFTDataset(Dataset):
+        def __init__(self, data, tokenizer):
+            self.input_ids = []
+            self.attn_masks = []
+            self.labels = []
+            self.prompts = []
+            EOS_ID = tokenizer("<|endoftext|>")["input_ids"][0]
```

Comment:
feel free to move this to after this PR, but changing this to tokenizer.eos_id would make it compatible with llama or others that don't use <|endoftext|>

Reply:
this applies to the other datasets and other <|endoftext|> instances

Reply:
Well, turns out the tokenizer does not always do EOS tokens, and will tokenize them assuming they aren't EOS (looking very angrily at you llama tokenizer)